source/Halibut/Transport/SecureListener.cs

```diff
@@ -399,9 +407,6 @@ static string GetThumbprint(SslStream stream)
 
         bool Authorize(string thumbprint, EndPoint clientName)
         {
-            if (thumbprint == null)
```

Comment:
Unnecessary guard, as we do this exact check on the line before we call `Authorize`.